profitbricks/src/main/java/org/jclouds/profitbricks/domain/Server.java

```diff
@@ -0,0 +1,424 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.domain;
+
+import org.jclouds.profitbricks.domain.internal.ServerCommonProperties;
+import com.google.auto.value.AutoValue;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import java.util.Date;
+import org.jclouds.javax.annotation.Nullable;
+
+@AutoValue
+public abstract class Server implements ServerCommonProperties {
+
+   public enum Status {
+
+      NOSTATE, RUNNING, BLOCKED, PAUSED, SHUTDOWN, SHUTOFF, CRASHED, UNRECOGNIZED;
+
+      public String value() {
+         return name();
+      }
+
+      public static Status fromValue( String v ) {
+         try {
+            return valueOf( v );
+         } catch ( IllegalArgumentException ex ) {
+            return UNRECOGNIZED;
+         }
+      }
+   }
+
+   @Nullable
+   public abstract String id();
+
+   @Nullable
+   @Override
+   public abstract String name();
+
+   @Nullable
+   public abstract boolean hasInternetAccess();
+
+   public abstract ProvisioningState state();
+
+   @Nullable
+   public abstract Status status();
+
+   @Nullable
+   public abstract OsType osType();
+
+   @Nullable
+   public abstract AvailabilityZone availabilityZone();
+
+   @Nullable
+   public abstract Date creationTime();
+
+   @Nullable
+   public abstract Date lastModificationTime();
+
+//   public abstract List<Storage> storages();
+//   public abstract List<Nic> storages();
+   public static Server create( String id, String name, int cores, int ram, boolean hasInternetAccess, ProvisioningState state,
+           Status status, OsType osType, AvailabilityZone availabilityZone, Date creationTime, Date lastModificationTime, Boolean isCpuHotPlug,
+           Boolean isRamHotPlug, Boolean isNicHotPlug, Boolean isNicHotUnPlug, Boolean isDiscVirtioHotPlug, Boolean isDiscVirtioHotUnPlug ) {
+      return new AutoValue_Server( isCpuHotPlug, isRamHotPlug, isNicHotPlug, isNicHotUnPlug, isDiscVirtioHotPlug, isDiscVirtioHotUnPlug,
+              cores, ram, id, name, hasInternetAccess, state, status, osType, availabilityZone, creationTime, lastModificationTime );
```

Comment:
Take care with inheritance, as the order here is very important, as noted in the [google auto warnings](https://github.com/google/auto/tree/master/value#warnings). Make sure we have a test to cover the proper object construction so we don't break this accidentally by changing the implemented interface.